remoteendpoint.go

```diff
@@ -94,7 +94,7 @@ func importRemoteEndpoints(sourceMap map[string]interface{}) ([]*remoteEndpoint,
 }
 
 func importRemoteEndpointList(sourceList []interface{}, importFunc remoteEndpointDeserializationFunc) ([]*remoteEndpoint, error) {
-	result := make([]*remoteEndpoint, 0, len(sourceList))
+	var result []*remoteEndpoint
```

Comment:
One of my tests showed that starting with a nil list of endpoints and roundtripping it would result in an empty slice, which seemed a bit nasty. This might be overly finicky. Or I could get the best of both worlds by adding an `if sourceList == nil { return nil, nil }` at the start.